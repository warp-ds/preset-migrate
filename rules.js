import {
  TYPES,
  COLOR_REGEXPS,
  COMING_SOON_MSG,
  emitWarning,
  CSS_DOCS_URL,
  COLOR_MESSAGES,
} from "./utils.js";

export default [
  [COLOR_REGEXPS.border, ([_]) => emitWarning(_, TYPES.replaced, COLOR_MESSAGES.border)],
  [COLOR_REGEXPS.background, ([_]) => emitWarning(_, TYPES.replaced, COLOR_MESSAGES.background)],
  [COLOR_REGEXPS.text, ([_]) => emitWarning(_, TYPES.replaced, COLOR_MESSAGES.text)],
  [/^text-(primary|secondary|danger)$/, ([_]) => emitWarning(_, TYPES.replaced, COLOR_MESSAGES.text)],
  [/^text-\d+$/, ([_]) => emitWarning(_, TYPES.replaced, `check ${CSS_DOCS_URL}/font-size for supported classes`)],
  [COLOR_REGEXPS.divide, ([_]) => emitWarning(_, TYPES.replaced, COMING_SOON_MSG)],
  [/^divide-(current|transparent|none)$/, ([_]) => emitWarning(_, TYPES.replaced, COMING_SOON_MSG)],
  [/^divide-(dotted|solid|double|dashed)$/, ([_]) => emitWarning(_, TYPES.removed)],
  [/^leading-\d+$/, ([_]) => emitWarning(_, TYPES.replaced, `check ${CSS_DOCS_URL}/line-height for supported classes`)],
  [/^aspect-[wh]-\d+$/, ([_]) => emitWarning(_, TYPES.replaced, "use fractions instead, e.g. aspect-4/3")],
  [/^flex-(shrink|grow)(-0)?$/, ([_, sg, d]) =>  emitWarning(_, TYPES.replaced, `use ${sg}${d || ""}`)],
  [/^shadow-none$/, ([_]) => emitWarning(_,TYPES.removed)],
  [/^shadow(-(2|3|4|10|20|30|40))?$/, ([_]) => emitWarning(_, TYPES.replaced, `check ${CSS_DOCS_URL}/box-shadow for supported shadow classes`)],
  [/^drop-shadow(-(2|3|4|10|20|30|40|none))?$/, ([_]) => emitWarning(_, TYPES.replaced, COMING_SOON_MSG)],
  [/^decoration-(slice|none|clone)/, ([_]) => emitWarning(_, TYPES.removed)],
  [/^ring(?:-.*)$/, ([_]) => emitWarning(_, TYPES.removed)],
  // filters
  [/^(backdrop-)?(filter|filter-none)$/, ([_]) => emitWarning(_, TYPES.removed)],
  [/^blur(-.*)?$/, ([_]) => emitWarning(_, TYPES.removed)],
  [/^backdrop-blur-(sm|md|lg|2xl|3xl)?$/, ([_]) => emitWarning(_, TYPES.replaced, 'Use sizes s|m|l|xxl|xxxl (instead of sm|md|lg|2xl|3xl)')],
  [/^(backdrop-)?(brightness|contrast|hue-rotate|saturate)-\d+$/, ([_]) => emitWarning(_, TYPES.removed)],
  [/^(backdrop-)?(invert|sepia)(-0)?$/, ([_]) => emitWarning(_, TYPES.removed)],
  [/^backdrop-opacity-\d+$/, ([_]) => emitWarning(_, TYPES.removed)],
  // outline
  [/^outline-(white|black)$/, ([_]) => emitWarning(_, TYPES.removed)],
  // semantic classes
  [/^(button|f)(-.+)?$/, ([_]) => emitWarning(_, TYPES.removed)],
  [/^input(--is-invalid|--is-disabled|--is-select__wrap|--is-read-only|-toggle)?$/, ([_]) => emitWarning(_, TYPES.removed)],
  [/^segment-control([-_].+)?$/,([_]) => emitWarning(_, TYPES.removed)],
  [/^field(-hint|-label)?$/,([_]) => emitWarning(_, TYPES.removed)],
  [/^link(--(block|dark))?$/,([_]) => emitWarning(_, TYPES.removed)],
  // old design system classes (troika)
  [/^t-.+$/, ([_]) => emitWarning(_, TYPES.removed)],
  // Do not target page-container, but container class with pseudo variants
  [/^([^page-].+)?container$/, ([_]) => emitWarning('container', TYPES.removed)],
  // miscellaneous
  [/^transition-gpu$/, ([_]) => emitWarning(_, TYPES.removed, "use 'transform-gpu' or 'will-change-*'")],
  [/^fixed-ios-fix$/,([_]) => emitWarning(_, TYPES.removed, "use 'transform translate-z-0'")],
  [/^focus-ring$/,  ([_]) => emitWarning(_, TYPES.replaced, "use 'focusable'")],
  [/^color-focused?$/, ([_]) => emitWarning(_, TYPES.removed)],
  [/^color-background(-.+)?$/, ([_]) => emitWarning(_, TYPES.removed, COLOR_MESSAGES.background)],
  [/^color-text(-.+)?$/, ([_]) => emitWarning(_, TYPES.removed, COLOR_MESSAGES.text)],

  // *** Warp v1 -> v2 ***
  // replaced semantic classes
  [/^(s-.+)-default$/, ([_, c]) => emitWarning(_, TYPES.replaced, `use '${c}'`)],
  // removed semantic classes
  [/^s-text-link-hover(-active)?$/, ([_]) => emitWarning(_, TYPES.removed, `use 's-text-link'`)],
  [/^(s-.+)-active-hover$/, ([_, c]) => emitWarning(_, TYPES.removed, `use '${c}-selected-hover' (if available)'`)],
  // removed internal classes
  [/^i-(bg|border|text|shadow)-\$(.*)$/, ([_, prop, token]) => emitWarning(_, TYPES.removed, `all internal classes (classes with 'i-' prefix) and most component specific tokens have been removed, ${COLOR_MESSAGES[prop==='bg'?'background':prop]} (or use class with arbitrary token value instead: '${prop}-[--w-${token}]', if this particular token still exists)`)],
  // removed alpha tokens
  [/^(.*--w-.*)-alpha(\d+)(.*)$/, ([_, prefix, alpha, suffix]) => emitWarning(_, TYPES.removed, `all 'alpha' tokens have been removed, try '${prefix}/${alpha}${suffix}', or reference ${CSS_DOCS_URL} for a suitable replacement`)],
  // removed primitive color tokens
  [/^.*--w-(bluegray|petroleum)-[1-9]\d{2,3}.*$/, ([_, color]) => emitWarning(_, TYPES.removed, `all '${color}' tokens have been removed, reference ${CSS_DOCS_URL} for a suitable replacement`)],
  // removed semantic color tokens without s-prefix
  [/^.*(--w-decoration-link)[^a-z-]*$/, ([_, token]) => emitWarning(_, TYPES.removed, `the '${token}' token has been removed`)],
  [/^.*--w-(color-background)(-subtle|)[^a-z-]*$/, ([_, token, tokenSuffix]) => emitWarning(_, TYPES.removed, `the '--w-${token}${tokenSuffix}' token has been removed. If applicable, use the 's-bg${tokenSuffix}' semantic class (or simply replace it with the equivalent semantic token '--w-s-${token}${tokenSuffix}')`)],
  [/^.*(--w-color-background-interactive(-subtle|-hover|))[^a-z-]*$/, ([_, token]) => emitWarning(_, TYPES.removed, `the '${token}' token has been removed`)],
  [/^.*--w-(color-text)(-subtle|-placeholder|-inverted(-subtle)*|-link|)[^a-z-]*$/, ([_, token, tokenSuffix]) => emitWarning(_, TYPES.removed, `the '--w-${token}${tokenSuffix}' token has been removed. If applicable, use the 's-text${tokenSuffix}' semantic class (or simply replace it with the equivalent semantic token '--w-s-${token}${tokenSuffix}')`)],
  [/^.*(--w-color-text-link(-hover|-visited))[^a-z-]*$/, ([_, token]) => emitWarning(_, TYPES.removed, `the '${token}' token has been removed`)],
  // removed semantic color tokens
  [/^.*(--w-s-color-background-(positive|negative|warning|info)-selected(-hover|-active)?)[^a-z-]*$/, ([_, token]) => emitWarning(_, TYPES.removed, `the '${token}' semantic token has been removed`)],
  [/^.*(--w-s-color-border-negative-selected(-hover)?)[^a-z-]*$/, ([_, token]) => emitWarning(_, TYPES.removed, `the '${token}' semantic token has been removed`)],
  // replaced semantic color tokens
  [/^.*(--w(-s)?-color-focused)[^a-z-].*$/, ([_, token]) => emitWarning(_, TYPES.replaced, `the '${token}' token has been replaced with '--w-s-border-focused'`)],
  [/^.*(--w-s-color-background(-subtle|-primary|-positive|-negative|-warning|-info|))-default[^a-z-].*$/, ([_, token]) => emitWarning(_, TYPES.replaced, `the '${token}-default' token has been replaced with '${token}'`)],
  [/^.*(--w-s-color-border((-primary|-positive|-negative|-warning|-info)(-subtle)?)?)-default[^a-z-].*$/, ([_, token]) => emitWarning(_, TYPES.replaced, `the '${token}-default' token has been replaced with '${token}'`)],
  [/^.*(--w-s-color-(text|icon(-subtle)?))-default[^a-z-].*$/, ([_, token]) => emitWarning(_, TYPES.replaced, `the '${token}-default' token has been replaced with '${token}'`)],
  // removed component color tokens
  [/^.*(--w-color-(alert|box|breadcrumbs|broadcast|card|checkbox|combobox|datepicker|expandable|helptext|image|input|label|list|modal|pageindicator|pagination|popover|radio|select|slider|spinner|starrating|stepindicator|tabs|toast|tooltip)(-[a-z-]*)?)[^a-z-]*$/, ([_, token]) => emitWarning(_, TYPES.removed, `the Warp internal '${token}' component token has been removed`)],
  [/^.*(--w-color-badge-(neutral|info|positive|warning|negative|disabled|price|notification)-text)[^a-z-]*$/, ([_, token]) => emitWarning(_, TYPES.removed, `the Warp internal '${token}' component token has been removed`)],
  [/^.*(--w-color-badge-(price|notification)-background)[^a-z-]*$/, ([_, token]) => emitWarning(_, TYPES.removed, `the Warp internal '${token}' component token has been removed`)],
  [/^.*(--w-color-button-primary-(text|icon|border(-hover|-active|)))[^a-z-]*$/, ([_, token]) => emitWarning(_, TYPES.removed, `the Warp internal '${token}' component token has been removed`)],
  [/^.*(--w-color-button-secondary-(text|icon|border(-hover|-active|)|background(-hover|-active|)))[^a-z-]*$/, ([_, token]) => emitWarning(_, TYPES.removed, `the Warp internal '${token}' component token has been removed`)],
  [/^.*(--w-color-button-quiet-(text|icon|background(-hover|-active|)))[^a-z-]*$/, ([_, token]) => emitWarning(_, TYPES.removed, `the Warp internal '${token}' component token has been removed`)],
  [/^.*(--w-color-button-negative(-quiet)?-(text|icon|border(-hover|-active|)|background(-hover|-active|)))[^a-z-]*$/, ([_, token]) => emitWarning(_, TYPES.removed, `the Warp internal '${token}' component token has been removed`)],
  [/^.*(--w-color-button-disabled-(text|background))[^a-z-]*$/, ([_, token]) => emitWarning(_, TYPES.removed, `the Warp internal '${token}' component token has been removed`)],
  [/^.*(--w-color-(button-link|callout)-text)[^a-z-]*$/, ([_, token]) => emitWarning(_, TYPES.removed, `the Warp internal '${token}' component token has been removed`)],
  [/^.*(--w-color-button-pill-(background|(overlay-)?icon(-hover|-active|)|))[^a-z-]*$/, ([_, token]) => emitWarning(_, TYPES.removed, `the Warp internal '${token}' component token has been removed`)],
  [/^.*(--w-color-button-loading-(text|background|icon))[^a-z-]*$/, ([_, token]) => emitWarning(_, TYPES.removed, `the Warp internal '${token}' component token has been removed`)],
  [/^.*(--w-color-button-utility(-quiet)?-(text|border(-hover|-active|)|background(-hover|-active|)))[^a-z-]*$/, ([_, token]) => emitWarning(_, TYPES.removed, `the Warp internal '${token}' component token has been removed`)],
  [/^.*(--w-color-buttongroup-primary-(text(-selected)?|background(-hover|-selected)?|border(-selected)?))[^a-z-]*$/, ([_, token]) => emitWarning(_, TYPES.removed, `the Warp internal '${token}' component token has been removed`)],
  [/^.*(--w-color-pill-(filter-(text|icon|background(-hover-|active|))|suggestion-(text|icon)))[^a-z-]*$/, ([_, token]) => emitWarning(_, TYPES.removed, `the Warp internal '${token}' component token has been removed`)],
  [/^.*(--w-color-switch-(track-background-(selected(-hover)?|disabled)|handle-background(-disabled)?))[^a-z-]*$/, ([_, token]) => emitWarning(_, TYPES.removed, `the Warp internal '${token}' component token has been removed`)],
  // removed component shadow tokens
  [/^.*(--w-shadow-(buttongroup|card(-hover)?|combobox|modal|popover|switch-handle|tooltip))[^a-z-]*$/, ([_, token]) => emitWarning(_, TYPES.removed, `the Warp internal '${token}' component token has been removed`)],
]
