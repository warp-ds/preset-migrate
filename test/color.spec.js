import { setup } from "./_helpers.js";
import { describe, expect, test, vi } from "vitest";
import { colors, colorHues } from "../rules.js"

setup();

describe("colors", () => {
  test("Emits a warning if text class is used with a color", async (t) => {
    const warnSpy = vi.spyOn(global.console, 'warn')

    const classes = colors.map((color) =>{
      return colorHues.map(hue => `text-${color}-${hue}`)
    }).flat()

    const { css } = await t.uno.generate(classes);

    expect(css).toMatchInlineSnapshot('""');
    expect(warnSpy).toHaveBeenCalledTimes(classes.length);
    expect(warnSpy.calls.flat()).toMatchSnapshot();
  });

  test("Emits a warning if border class is used with a color", async (t) => {
    const warnSpy = vi.spyOn(global.console, 'warn')

    const classes = colors.map((color) =>{
      return colorHues.map(hue => `border-${color}-${hue}`)
    }).flat()

    const { css } = await t.uno.generate(classes);

    expect(css).toMatchInlineSnapshot('""');
    expect(warnSpy).toHaveBeenCalledTimes(classes.length);
    expect(warnSpy.calls.flat()).toMatchSnapshot();
  });

  test("Emits a warning if background class is used with a color", async (t) => {
    const warnSpy = vi.spyOn(global.console, 'warn')

    const classes = colors.map((color) =>{
      return colorHues.map(hue => `bg-${color}-${hue}`)
    }).flat()

    const { css } = await t.uno.generate(classes);

    expect(css).toMatchInlineSnapshot('""');
    expect(warnSpy).toHaveBeenCalledTimes(classes.length);
    expect(warnSpy.calls.flat()).toMatchSnapshot();
  });

  test("Emits a warning if divide class is used with a color", async (t) => {
    const warnSpy = vi.spyOn(global.console, 'warn')

    const classes = colors.map((color) =>{
      return colorHues.map(hue => `divide-${color}-${hue}`)
    }).flat()

    const { css } = await t.uno.generate(classes);

    expect(css).toMatchInlineSnapshot('""');
    expect(warnSpy).toHaveBeenCalledTimes(classes.length);
    expect(warnSpy.calls.flat()).toMatchSnapshot();
  });

  test("Emits a warning if 'text', 'bg', 'border' or 'divide' is used with 'current', 'transparent', 'none' or 'white'", async (t) => {
    const warnSpy = vi.spyOn(global.console, 'warn')

    const classes = ["current", "transparent", "none", "white"].map((color) =>{
      return ["text", "bg", "border", "divide"].map(el => `${el}-${color}`)
    }).flat()

    const { css } = await t.uno.generate(classes);

    expect(css).toMatchInlineSnapshot('""');
    expect(warnSpy).toHaveBeenCalledTimes(classes.length);
    expect(warnSpy.calls.flat()).toMatchSnapshot();
  });

  test("Emits a warning for color classes with pseudo", async (t) => {
    const warnSpy = vi.spyOn(global.console, 'warn')

    const classes = ["focus:text-blue-500", "active:divide-yellow-100", "sm:border-green-50", "!bg-yellow-400"]

    const { css } = await t.uno.generate(classes);

    expect(css).toMatchInlineSnapshot('""');
    expect(warnSpy).toHaveBeenCalledTimes(classes.length);
    expect(warnSpy.calls.flat()).toMatchSnapshot();
  });
})