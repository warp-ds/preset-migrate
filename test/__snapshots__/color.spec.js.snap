// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`colors > Emits a warning for color classes with pseudo 2`] = `
[
  "[REPLACED] text-blue-500",
  "[REPLACED] divide-yellow-100",
  "[REPLACED] border-green-50",
  "[REPLACED] bg-yellow-400",
]
`;

exports[`colors > Emits a warning if 'bg' is used with 'none' or 'white' 2`] = `
[
  "[REPLACED] bg-none",
  "[REPLACED] bg-white",
]
`;

exports[`colors > Emits a warning if 'text', 'border' or 'divide' is used with 'current', 'transparent', 'none' or 'white' 2`] = `
[
  "[REPLACED] text-current",
  "[REPLACED] border-current",
  "[REPLACED] divide-current",
  "[REPLACED] text-transparent",
  "[REPLACED] border-transparent",
  "[REPLACED] divide-transparent",
  "[REPLACED] text-none",
  "[REPLACED] border-none",
  "[REPLACED] divide-none",
  "[REPLACED] text-white",
  "[REPLACED] border-white",
  "[REPLACED] divide-white",
]
`;

exports[`colors > Emits a warning if background class is used with a color 2`] = `
[
  "[REPLACED] bg-blue-50",
  "[REPLACED] bg-blue-100",
  "[REPLACED] bg-blue-200",
  "[REPLACED] bg-blue-300",
  "[REPLACED] bg-blue-400",
  "[REPLACED] bg-blue-500",
  "[REPLACED] bg-blue-600",
  "[REPLACED] bg-blue-700",
  "[REPLACED] bg-blue-800",
  "[REPLACED] bg-blue-900",
  "[REPLACED] bg-green-50",
  "[REPLACED] bg-green-100",
  "[REPLACED] bg-green-200",
  "[REPLACED] bg-green-300",
  "[REPLACED] bg-green-400",
  "[REPLACED] bg-green-500",
  "[REPLACED] bg-green-600",
  "[REPLACED] bg-green-700",
  "[REPLACED] bg-green-800",
  "[REPLACED] bg-green-900",
  "[REPLACED] bg-aqua-50",
  "[REPLACED] bg-aqua-100",
  "[REPLACED] bg-aqua-200",
  "[REPLACED] bg-aqua-300",
  "[REPLACED] bg-aqua-400",
  "[REPLACED] bg-aqua-500",
  "[REPLACED] bg-aqua-600",
  "[REPLACED] bg-aqua-700",
  "[REPLACED] bg-aqua-800",
  "[REPLACED] bg-aqua-900",
  "[REPLACED] bg-yellow-50",
  "[REPLACED] bg-yellow-100",
  "[REPLACED] bg-yellow-200",
  "[REPLACED] bg-yellow-300",
  "[REPLACED] bg-yellow-400",
  "[REPLACED] bg-yellow-500",
  "[REPLACED] bg-yellow-600",
  "[REPLACED] bg-yellow-700",
  "[REPLACED] bg-yellow-800",
  "[REPLACED] bg-yellow-900",
  "[REPLACED] bg-red-50",
  "[REPLACED] bg-red-100",
  "[REPLACED] bg-red-200",
  "[REPLACED] bg-red-300",
  "[REPLACED] bg-red-400",
  "[REPLACED] bg-red-500",
  "[REPLACED] bg-red-600",
  "[REPLACED] bg-red-700",
  "[REPLACED] bg-red-800",
  "[REPLACED] bg-red-900",
  "[REPLACED] bg-bluegray-50",
  "[REPLACED] bg-bluegray-100",
  "[REPLACED] bg-bluegray-200",
  "[REPLACED] bg-bluegray-300",
  "[REPLACED] bg-bluegray-400",
  "[REPLACED] bg-bluegray-500",
  "[REPLACED] bg-bluegray-600",
  "[REPLACED] bg-bluegray-700",
  "[REPLACED] bg-bluegray-800",
  "[REPLACED] bg-bluegray-900",
  "[REPLACED] bg-gray-50",
  "[REPLACED] bg-gray-100",
  "[REPLACED] bg-gray-200",
  "[REPLACED] bg-gray-300",
  "[REPLACED] bg-gray-400",
  "[REPLACED] bg-gray-500",
  "[REPLACED] bg-gray-600",
  "[REPLACED] bg-gray-700",
  "[REPLACED] bg-gray-800",
  "[REPLACED] bg-gray-900",
]
`;

exports[`colors > Emits a warning if border class is used with a color 2`] = `
[
  "[REPLACED] border-blue-50",
  "[REPLACED] border-blue-100",
  "[REPLACED] border-blue-200",
  "[REPLACED] border-blue-300",
  "[REPLACED] border-blue-400",
  "[REPLACED] border-blue-500",
  "[REPLACED] border-blue-600",
  "[REPLACED] border-blue-700",
  "[REPLACED] border-blue-800",
  "[REPLACED] border-blue-900",
  "[REPLACED] border-green-50",
  "[REPLACED] border-green-100",
  "[REPLACED] border-green-200",
  "[REPLACED] border-green-300",
  "[REPLACED] border-green-400",
  "[REPLACED] border-green-500",
  "[REPLACED] border-green-600",
  "[REPLACED] border-green-700",
  "[REPLACED] border-green-800",
  "[REPLACED] border-green-900",
  "[REPLACED] border-aqua-50",
  "[REPLACED] border-aqua-100",
  "[REPLACED] border-aqua-200",
  "[REPLACED] border-aqua-300",
  "[REPLACED] border-aqua-400",
  "[REPLACED] border-aqua-500",
  "[REPLACED] border-aqua-600",
  "[REPLACED] border-aqua-700",
  "[REPLACED] border-aqua-800",
  "[REPLACED] border-aqua-900",
  "[REPLACED] border-yellow-50",
  "[REPLACED] border-yellow-100",
  "[REPLACED] border-yellow-200",
  "[REPLACED] border-yellow-300",
  "[REPLACED] border-yellow-400",
  "[REPLACED] border-yellow-500",
  "[REPLACED] border-yellow-600",
  "[REPLACED] border-yellow-700",
  "[REPLACED] border-yellow-800",
  "[REPLACED] border-yellow-900",
  "[REPLACED] border-red-50",
  "[REPLACED] border-red-100",
  "[REPLACED] border-red-200",
  "[REPLACED] border-red-300",
  "[REPLACED] border-red-400",
  "[REPLACED] border-red-500",
  "[REPLACED] border-red-600",
  "[REPLACED] border-red-700",
  "[REPLACED] border-red-800",
  "[REPLACED] border-red-900",
  "[REPLACED] border-bluegray-50",
  "[REPLACED] border-bluegray-100",
  "[REPLACED] border-bluegray-200",
  "[REPLACED] border-bluegray-300",
  "[REPLACED] border-bluegray-400",
  "[REPLACED] border-bluegray-500",
  "[REPLACED] border-bluegray-600",
  "[REPLACED] border-bluegray-700",
  "[REPLACED] border-bluegray-800",
  "[REPLACED] border-bluegray-900",
  "[REPLACED] border-gray-50",
  "[REPLACED] border-gray-100",
  "[REPLACED] border-gray-200",
  "[REPLACED] border-gray-300",
  "[REPLACED] border-gray-400",
  "[REPLACED] border-gray-500",
  "[REPLACED] border-gray-600",
  "[REPLACED] border-gray-700",
  "[REPLACED] border-gray-800",
  "[REPLACED] border-gray-900",
]
`;

exports[`colors > Emits a warning if divide class is used with a color 2`] = `
[
  "[REPLACED] divide-blue-50",
  "[REPLACED] divide-blue-100",
  "[REPLACED] divide-blue-200",
  "[REPLACED] divide-blue-300",
  "[REPLACED] divide-blue-400",
  "[REPLACED] divide-blue-500",
  "[REPLACED] divide-blue-600",
  "[REPLACED] divide-blue-700",
  "[REPLACED] divide-blue-800",
  "[REPLACED] divide-blue-900",
  "[REPLACED] divide-green-50",
  "[REPLACED] divide-green-100",
  "[REPLACED] divide-green-200",
  "[REPLACED] divide-green-300",
  "[REPLACED] divide-green-400",
  "[REPLACED] divide-green-500",
  "[REPLACED] divide-green-600",
  "[REPLACED] divide-green-700",
  "[REPLACED] divide-green-800",
  "[REPLACED] divide-green-900",
  "[REPLACED] divide-aqua-50",
  "[REPLACED] divide-aqua-100",
  "[REPLACED] divide-aqua-200",
  "[REPLACED] divide-aqua-300",
  "[REPLACED] divide-aqua-400",
  "[REPLACED] divide-aqua-500",
  "[REPLACED] divide-aqua-600",
  "[REPLACED] divide-aqua-700",
  "[REPLACED] divide-aqua-800",
  "[REPLACED] divide-aqua-900",
  "[REPLACED] divide-yellow-50",
  "[REPLACED] divide-yellow-100",
  "[REPLACED] divide-yellow-200",
  "[REPLACED] divide-yellow-300",
  "[REPLACED] divide-yellow-400",
  "[REPLACED] divide-yellow-500",
  "[REPLACED] divide-yellow-600",
  "[REPLACED] divide-yellow-700",
  "[REPLACED] divide-yellow-800",
  "[REPLACED] divide-yellow-900",
  "[REPLACED] divide-red-50",
  "[REPLACED] divide-red-100",
  "[REPLACED] divide-red-200",
  "[REPLACED] divide-red-300",
  "[REPLACED] divide-red-400",
  "[REPLACED] divide-red-500",
  "[REPLACED] divide-red-600",
  "[REPLACED] divide-red-700",
  "[REPLACED] divide-red-800",
  "[REPLACED] divide-red-900",
  "[REPLACED] divide-bluegray-50",
  "[REPLACED] divide-bluegray-100",
  "[REPLACED] divide-bluegray-200",
  "[REPLACED] divide-bluegray-300",
  "[REPLACED] divide-bluegray-400",
  "[REPLACED] divide-bluegray-500",
  "[REPLACED] divide-bluegray-600",
  "[REPLACED] divide-bluegray-700",
  "[REPLACED] divide-bluegray-800",
  "[REPLACED] divide-bluegray-900",
  "[REPLACED] divide-gray-50",
  "[REPLACED] divide-gray-100",
  "[REPLACED] divide-gray-200",
  "[REPLACED] divide-gray-300",
  "[REPLACED] divide-gray-400",
  "[REPLACED] divide-gray-500",
  "[REPLACED] divide-gray-600",
  "[REPLACED] divide-gray-700",
  "[REPLACED] divide-gray-800",
  "[REPLACED] divide-gray-900",
]
`;

exports[`colors > Emits a warning if text class is used with a color 2`] = `
[
  "[REPLACED] text-blue-50",
  "[REPLACED] text-blue-100",
  "[REPLACED] text-blue-200",
  "[REPLACED] text-blue-300",
  "[REPLACED] text-blue-400",
  "[REPLACED] text-blue-500",
  "[REPLACED] text-blue-600",
  "[REPLACED] text-blue-700",
  "[REPLACED] text-blue-800",
  "[REPLACED] text-blue-900",
  "[REPLACED] text-green-50",
  "[REPLACED] text-green-100",
  "[REPLACED] text-green-200",
  "[REPLACED] text-green-300",
  "[REPLACED] text-green-400",
  "[REPLACED] text-green-500",
  "[REPLACED] text-green-600",
  "[REPLACED] text-green-700",
  "[REPLACED] text-green-800",
  "[REPLACED] text-green-900",
  "[REPLACED] text-aqua-50",
  "[REPLACED] text-aqua-100",
  "[REPLACED] text-aqua-200",
  "[REPLACED] text-aqua-300",
  "[REPLACED] text-aqua-400",
  "[REPLACED] text-aqua-500",
  "[REPLACED] text-aqua-600",
  "[REPLACED] text-aqua-700",
  "[REPLACED] text-aqua-800",
  "[REPLACED] text-aqua-900",
  "[REPLACED] text-yellow-50",
  "[REPLACED] text-yellow-100",
  "[REPLACED] text-yellow-200",
  "[REPLACED] text-yellow-300",
  "[REPLACED] text-yellow-400",
  "[REPLACED] text-yellow-500",
  "[REPLACED] text-yellow-600",
  "[REPLACED] text-yellow-700",
  "[REPLACED] text-yellow-800",
  "[REPLACED] text-yellow-900",
  "[REPLACED] text-red-50",
  "[REPLACED] text-red-100",
  "[REPLACED] text-red-200",
  "[REPLACED] text-red-300",
  "[REPLACED] text-red-400",
  "[REPLACED] text-red-500",
  "[REPLACED] text-red-600",
  "[REPLACED] text-red-700",
  "[REPLACED] text-red-800",
  "[REPLACED] text-red-900",
  "[REPLACED] text-bluegray-50",
  "[REPLACED] text-bluegray-100",
  "[REPLACED] text-bluegray-200",
  "[REPLACED] text-bluegray-300",
  "[REPLACED] text-bluegray-400",
  "[REPLACED] text-bluegray-500",
  "[REPLACED] text-bluegray-600",
  "[REPLACED] text-bluegray-700",
  "[REPLACED] text-bluegray-800",
  "[REPLACED] text-bluegray-900",
  "[REPLACED] text-gray-50",
  "[REPLACED] text-gray-100",
  "[REPLACED] text-gray-200",
  "[REPLACED] text-gray-300",
  "[REPLACED] text-gray-400",
  "[REPLACED] text-gray-500",
  "[REPLACED] text-gray-600",
  "[REPLACED] text-gray-700",
  "[REPLACED] text-gray-800",
  "[REPLACED] text-gray-900",
]
`;
